File: programming-concepts/good-programming-practice/visual-basic/identifiers.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-SA 4
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    Const GravityOnEarth As Double = 9.81
    Const GravityOnMars As Double = 3.711

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim weight As Double = 35.7
        Console.WriteLine(Conv(weight))
        Console.WriteLine(CalculateWeightOnMars(weight))
    End Sub

    Function Conv(ByVal a As Double) As Double
        Dim b As Double = a / 9.81 * 3.711
        Return b
    End Function

    Function CalculateWeightOnMars(ByVal weightOnEarth As Double) As Double
        Dim weightOnMars As Double = weightOnEarth / GravityOnEarth * GravityOnMars
        Return weightOnMars
    End Function

End Module

File: programming-concepts/good-programming-practice/python/identifiers.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


# First example
def conv(a):
   b = a / 9.81 * 3.711
   return b


# Second example
GRAVITY_ON_EARTH = 9.81
GRAVITY_ON_MARS = 3.711

def calculate_weight_on_mars(weight_on_earth):
   weight_on_mars = weight_on_earth / GRAVITY_ON_EARTH * GRAVITY_ON_MARS
   return weight_on_mars


def main():
    """Execute both subroutines and output the results"""

    weight = 35.7
    print(conv(weight))
    print(calculate_weight_on_mars(weight))


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()
